lib/cnet/ipv4/ip4_input.c

```diff
@@ -194,9 +208,12 @@ ip4_input_node_process(struct cne_graph *graph, struct cne_node *node, void **ob
         n_left_from -= 1;
 
         /* Extract DIP from mbuf */
-        ip4[0] = pktmbuf_adjust(mbuf0, struct cne_ipv4_hdr *, mbuf0->l2_len);
+        ip4[0] = pktmbuf_mtod(mbuf0, struct cne_ipv4_hdr *);
         dip[0] = be32toh(ip4[0]->dst_addr);
 
+        /* Adjust the data length for an IPv4 packet to the size given in the header */
+        pktmbuf_data_len(mbuf0) = be16toh(ip4[0]->total_length);
```

Comment:
Need the length check here too

Reply:
Fixed, which it appears I forgot to do the checksum too.